src/utils/env.js

```diff
@@ -23,68 +23,70 @@ const homedir = os.homedir();
 const configDirPath = `${homedir}/${configDirName}`;
 const configFilePath = `${configDirPath}/${configFileName}`;
 
-const getReadErrorMessage = (error, path) => {
-	if (error.code === 'EACCES') {
-		return `WARNING: Could not read \`${path}\`: permission denied. Using default config instead.`;
-	}
-	if (error.message.match(/JSON/)) {
-		return `WARNING: Config file at \`${path}\` is not valid JSON. Using default config instead.`;
+const getWriteErrorMessage = path => `WARNING: Could not write to \`${path}\`. Your configuration will not be persisted.`;
+
+const attemptCallWithWarning = (fn, path) => {
+	try {
+		return fn();
+	} catch (_) {
+		const warning = getWriteErrorMessage(path);
+		console.warn(warning);
+		return null;
 	}
-	return `WARNING: Could not read \`${path}\`. Using default config instead.`;
 };
 
-const getWriteErrorMessage = path => `WARNING: Could not write to \`${path}\`. Your configuration will not be persisted.`;
-
-const checkExists = (path) => {
+const attemptCallWithError = (fn, code, error) => {
 	try {
-		return fse.existsSync(path);
-	} catch (error) {
-		const message = getReadErrorMessage(error, path);
-		console.warn(message);
-		return false;
+		return fn();
+	} catch (_) {
+		console.error(error);
+		return process.exit(code);
 	}
 };
 
-const initConfigFile = () => {
-	const configDirExists = checkExists(configDirPath);
+const checkWriteAccess = (path) => {
```

Comment:
This should not happen on default startup but on every time we try to write to the file in set command.